torch/_inductor/pcache.py

```diff
@@ -0,0 +1,317 @@
+from concurrent.futures import Future, ThreadPoolExecutor
+from functools import cached_property
+from os import getenv
+from pathlib import Path
+from typing_extensions import Self
+
+from torch.utils._filelock import FileLock
+
+
+# Key represents data used to query from cache instances
+Key = str
```

Comment:
Please type these as typing_extensions.TypeAlias to make it easier for static analysis